cfme/infrastructure/config_management.py

```diff
@@ -0,0 +1,243 @@
+from functools import partial
+
+import ui_navigate as nav
+
+import cfme
+import cfme.fixtures.pytest_selenium as sel
+import cfme.web_ui.flash as flash
+import cfme.web_ui.menu
+import cfme.web_ui.toolbar as tb
+from cfme.web_ui import (
+    accordion, Quadicon, Form, Input, fill, form_buttons, toolbar, SplitTable
+)
+from utils import lazycache, version, conf
+from utils.pretty import Pretty
+from utils.update import Updateable
+from utils.wait import wait_for
+
+
+properties_form = Form(
+    fields=[
+        ('name_text', Input('name')),
+        ('url_text', Input('url')),
+        ('ssl_checkbox', Input('verify_ssl'))
+    ])
+
+credential_form = Form(
+    fields=[
+        ('principal_text', Input('log_userid')),
+        ('secret_pass', Input('log_password')),
+        ('verify_secret_pass', Input('log_verify')),
+        ('validate_btn', form_buttons.validate)
+    ])
+
+list_table = SplitTable(
+    header_data=("//div[@id='list_grid']/div[@class='xhdr']/table/tbody", 1),
+    body_data=("//div[@id='list_grid']/div[@class='objbox']/table/tbody", 1),
+)
+
+add_manager_btn = form_buttons.FormButton('Add')
+cfg_btn = partial(tb.select, 'Configuration')
+
+nav.add_branch(
+    'infrastructure_config_management',
+    {
+        'infrastructure_config_managers':
+        [
+            lambda _: accordion.tree('Providers',
+                version.pick({version.LOWEST: 'All Red Hat Satellite Providers',
+                              version.LATEST: 'All Foreman Providers'})),
```

Comment:
I also suggested a possibility of a ternary expression with `is_appliance_downstream` but I am not sure if that is the best approach.